remote-trait-object/src/packet.rs

```diff
@@ -0,0 +1,230 @@
+// Copyright 2020 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use std::fmt;
+
+pub struct SlotId(u32);
+
+impl fmt::Debug for SlotId {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("SlotId").field("id", &self.0).field("type", &self.get_type()).finish()
```

Comment:
https://doc.rust-lang.org/std/fmt/trait.Debug.html
"Generally speaking, you should just derive a Debug implementation."
I think this kind of implementation would belong to `Display`.

Reply:
Thanks. I chaged it.